homeassistant/components/vallox/sensor.py

```diff
@@ -21,170 +27,23 @@
-            metric_key="A_CYC_TEMP_EXTRACT_AIR",
-            device_class=DEVICE_CLASS_TEMPERATURE,
-            unit_of_measurement=TEMP_CELSIUS,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxSensor(
-            name=f"{name} Exhaust Air",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_TEMP_EXHAUST_AIR",
-            device_class=DEVICE_CLASS_TEMPERATURE,
-            unit_of_measurement=TEMP_CELSIUS,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxSensor(
-            name=f"{name} Outdoor Air",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_TEMP_OUTDOOR_AIR",
-            device_class=DEVICE_CLASS_TEMPERATURE,
-            unit_of_measurement=TEMP_CELSIUS,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxSensor(
-            name=f"{name} Supply Air",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_TEMP_SUPPLY_AIR",
-            device_class=DEVICE_CLASS_TEMPERATURE,
-            unit_of_measurement=TEMP_CELSIUS,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxSensor(
-            name=f"{name} Humidity",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_RH_VALUE",
-            device_class=DEVICE_CLASS_HUMIDITY,
-            unit_of_measurement=PERCENTAGE,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxFilterRemainingSensor(
-            name=f"{name} Remaining Time For Filter",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_REMAINING_TIME_FOR_FILTER",
-            device_class=DEVICE_CLASS_TIMESTAMP,
-            unit_of_measurement=None,
-            icon="mdi:filter",
-        ),
-        ValloxSensor(
-            name=f"{name} Efficiency",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_EXTRACT_EFFICIENCY",
-            device_class=None,
-            unit_of_measurement=PERCENTAGE,
-            icon="mdi:gauge",
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-        ValloxSensor(
-            name=f"{name} CO2",
-            state_proxy=state_proxy,
-            metric_key="A_CYC_CO2_VALUE",
-            device_class=DEVICE_CLASS_CO2,
-            unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
-            icon=None,
-            state_class=STATE_CLASS_MEASUREMENT,
-        ),
-    ]
-
-    async_add_entities(sensors, update_before_add=False)
-
-
 class ValloxSensor(SensorEntity):
     """Representation of a Vallox sensor."""
 
+    _attr_should_poll = False
+
     def __init__(
-        self,
-        name,
-        state_proxy,
-        metric_key,
-        device_class,
-        unit_of_measurement,
-        icon,
-        state_class=None,
+        self, name, state_proxy, description: ValloxSensorEntityDescription
```

Comment:
Don't add partial typing to methods, add for name and state_proxy as well :+1:

Reply:
Thanks, done.